core/shared/src/main/scala/scalaz/zio/stream/Sink.scala

```diff
@@ -428,13 +441,13 @@ object Sink {
 
   type Step[+S, +A0] = Step.Step[S, A0]
 
-  final def more[R, E, A0, A, B](end: IO[E, B])(input: A => Sink[R, E, A0, A, B]): Sink[R, E, A0, A, B] =
-    new Sink[R, E, A0, A, B] {
-      type State = Option[(Sink[R, E, A0, A, B], Any)]
+  final def more[R, R1 <: R, E, A0, A, B](end: ZIO[R1, E, B])(input: A => Sink[R, E, A0, A, B]): Sink[R1, E, A0, A, B] =
```

Comment:
`input: A => Sink[R1, E, A0, A, B]` ?

Reply:
So the input function yields sinks that use `R`, and the ending value `end` uses `R1`